optimum/neuron/modeling_diffusion.py

```diff
@@ -842,9 +850,20 @@ def __init__(
     ):
         super().__init__(model, parent_model, config, neuron_config, DIFFUSION_MODEL_TEXT_ENCODER_NAME)
 
-    def forward(self, input_ids: torch.Tensor):
+    def forward(
+        self,
+        input_ids: torch.Tensor,
+        *args,
+        **kwargs,
+    ):
+        input_ids = input_ids.to(torch.long)  # dummy generator uses long int for tracing
+
         inputs = (input_ids,)
         outputs = self.model(*inputs)
+
+        if kwargs.pop("return_dict", False):
```

Comment:
supplementing the suggestion above so that you can directly apply and commit
```suggestion
        if return_dict:
```